### PR TITLE
LINT: Replace try..close with try-with-resources

### PR DESCRIPTION
**Pull request by david-allison-1**
## Purpose / Description

Reduce amount of boilerplate code in project

Available on all API levels:

> In addition to the Java 8 language features and APIs above, Android Studio 3.0 and later extends support for try-with-resources to all Android API levels.

https://developer.android.com/studio/write/java8-support

## Fixes
N/A

## Approach

Automated lint fix from Android Studio (except 1, which involved moving a nullvariable, then an auto fix)

Finally, manual undo of the whitespace changes.

## How Has This Been Tested?

Unit, Android, and tested on my machine

## Learning (optional, can help others)
https://developer.android.com/studio/write/java8-support

## Checklist
- [x] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code

### CODE REVIEW

**Comment by timrae** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity decreasing per file
==============================
+ AnkiDroid/src/main/java/com/ichi2/libanki/Tags.java  -1
+ AnkiDroid/src/main/java/com/ichi2/anki/multimediacard/fields/BasicAudioClipFieldController.java  -2
+ AnkiDroid/src/main/java/com/ichi2/libanki/DB.java  -1
+ AnkiDroid/src/main/java/com/ichi2/libanki/importer/Anki2Importer.java  -1
+ AnkiDroid/src/main/java/com/ichi2/libanki/Media.java  -1
+ AnkiDroid/src/main/java/com/ichi2/utils/ImportUtils.java  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/timrae/Anki-Android/pullRequest?prid=5344894&bid=17499816)

File: AnkiDroid/src/main/java/com/ichi2/anki/multimediacard/fields/BasicAudioClipFieldController.java [hunk 1]
```diff
@@ -147,15 +145,6 @@ public void onActivityResult(int requestCode, int resultCode, Intent data) {
                 UIUtils.showThemedToast(AnkiDroidApp.getInstance().getApplicationContext(),
                         AnkiDroidApp.getInstance().getString(R.string.multimedia_editor_something_wrong), true);
             }
-            finally {
-                try {
-                    if (inputStream != null) {
-                        inputStream.close();
```
Comment (mikehardy) [inline AnkiDroid/src/main/java/com/ichi2/anki/multimediacard/fields/BasicAudioClipFieldController.java:153]: I double-checked and close exceptions in try-with-resources are suppressed in that you can't handle them, but they are caught and logged which is the same "recovery" (or lack of) we are attempting. :+1:

**Comment by timrae** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity decreasing per file
==============================
+ AnkiDroid/src/main/java/com/ichi2/libanki/Tags.java  -1
+ AnkiDroid/src/main/java/com/ichi2/libanki/DB.java  -1
+ AnkiDroid/src/main/java/com/ichi2/libanki/Media.java  -1
+ AnkiDroid/src/main/java/com/ichi2/libanki/importer/Anki2Importer.java  -1
+ AnkiDroid/src/main/java/com/ichi2/anki/multimediacard/fields/BasicAudioClipFieldController.java  -2
+ AnkiDroid/src/main/java/com/ichi2/utils/ImportUtils.java  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/timrae/Anki-Android/pullRequest?prid=5344894&bid=17499816)

File: AnkiDroid/src/main/java/com/ichi2/libanki/Card.java [hunk 2]
```diff
@@ -163,10 +161,6 @@ public void load() {
             mODid = cursor.getLong(15);
             mFlags = cursor.getInt(16);
             mData = cursor.getString(17);
-        } finally {
-            if (cursor != null) {
-                cursor.close();
```
Comment (mikehardy) [inline AnkiDroid/src/main/java/com/ichi2/libanki/Card.java:168]: And this could have thrown I think, so this will actually / maybe fix some issues

**Comment by mikehardy** [comment]
out as 2.10alpha68 as soon as google is done processing it...

**Pull request merged**